ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +345,348 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
+        assert.equal(
+          component.state.name,
+          ''
+        );
+        assert.equal(
+          component.state.description,
+          ''
+        );
+        assert.equal(
+          component.state.vocabularyType,
+          'm'
+        );
+        assert.equal(
+          component.state.learningResourceTypes.length,
+          0
+        );
+        var formNode = assertAddVocabularyCommon(assert, component);
+        waitForAjax(1, function() {
+          assert.equal(
+            component.state.vocabularyType,
+            'm'
+          );
+          assert.equal(
+            component.state.learningResourceTypes.length,
+            0
+          );
+          assert.equal(
+            saveVocabularyResponse.id,
+            vocabulary.id
+          );
+          var inputNodes = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithTag(
+            formNode,
+            'input'
+          );
+          inputNodes[0].value = "TestA";
+          inputNodes[1].value = "TestA";
+          inputNodes[2].checked = true;
+          React.addons.TestUtils.Simulate.change(
+              inputNodes[2],
+              {target: {value: 'course'}}
+            );
+          inputNodes[11].checked = true;
+          React.addons.TestUtils.Simulate.change(
+              inputNodes[11],
+              {target: {value: 'f'}}
+            );
+          React.addons.TestUtils.Simulate.submit(formNode);
+          waitForAjax(1, function() {
+            assert.equal(
+              saveVocabularyResponse.learning_resource_types[0],
+              "course"
+            );
+            assert.equal(
+              saveVocabularyResponse.id,
+              vocabulary.id
+            );
+            done();
+          });
+        });
+      };
+      React.addons.TestUtils.
+        renderIntoDocument(
+          <AddVocabulary
+            vocabularies={vocabularies}
+            repoSlug="repo"
+            updateParent={updateParent}
+            learningResourceTypes={learningResourceTypes}
+            ref={afterMount}
+          />
+        );
+    }
+  );
+
+  QUnit.test('Assert that ajax fail in AddVocabulary',
+    function(assert) {
+      assert.ok(AddVocabulary, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var done = assert.async();
+      var updateParent = function() {};
+      var afterMount = function(component) {
+        var formNode = React.addons.TestUtils.
+          findRenderedDOMComponentWithClass(
+            component,
+            'form-horizontal'
+          );
+        assert.ok(formNode);
+        //test form submission
+        var inputNodes = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithTag(
+            formNode,
+            'input'
+          );
+        assert.equal(inputNodes.length, 4);
+        inputNodes[0].value = "TestA";
```

Comment:
We don't need to set any of these since we're just getting mocked AJAX back anyway